src/voodoo-gen/search_index.ml

```diff
@@ -96,6 +113,17 @@ module Generate = struct
           :: full_name_aux (parent :> Identifier.t)
       | `Label (parent, name) ->
           LabelName.to_string name :: full_name_aux (parent :> Identifier.t)
+      | `SourceDir (parent, name) ->
+          name :: full_name_aux (parent :> Identifier.t)
+      | `AssetFile (parent, name) ->
+          name :: full_name_aux (parent :> Identifier.t)
+      | `SourceLocationMod parent -> full_name_aux (parent :> Identifier.t)
+      | `SourceLocation (parent, name) ->
+          DefName.to_string name :: full_name_aux (parent :> Identifier.t)
+      | `SourceLocationInternal (parent, name) ->
+          LocalName.to_string name :: full_name_aux (parent :> Identifier.t)
+      | `SourcePage (parent, name) ->
+          name :: full_name_aux (parent :> Identifier.t)
```

Comment:
This is fine, although a fullname function is now shipped by odoc, so it could be deleted.